### PR TITLE
Update type definitions

### PR DESCRIPTION
**Pull request by pbadenski**
Mostly updating after #137 and #142 changes

Additionally add:
 - sampler namespace
 - Record interface
 - TraceId properties

### CODE REVIEW

File: packages/zipkin/index.d.ts [hunk 1]
```diff
@@ -85,9 +112,48 @@ declare namespace zipkin {
     }
   }
 
+  namespace model {
+    interface Endpoint {
+      constructor(args: { serviceName?: string, ipv4?: InetAddress, port?: number });
+
+      setServiceName(serviceName: string): void;
+      setIpv4(ipv4: InetAddress): void;
```
Comment (adriancole) [inline packages/zipkin/index.d.ts]: technically this is a string literal which has an ipv4 inside..

File: packages/zipkin/index.d.ts [hunk 2]
```diff
@@ -85,9 +112,48 @@ declare namespace zipkin {
...
+      traceId:        option.IOption<string>;
+      spanId:         string;
+      name:           string;
+      kind:           string;
+      timestamp:      number;
+      duration:       number;
+      localEndpoint:  Endpoint;
+      remoteEndpoint: Endpoint;
+      annotations:    any[];
+      tags:           any;
```
Comment (adriancole) [inline packages/zipkin/index.d.ts]: is there a way to say dict string -> string?

File: packages/zipkin/index.d.ts [hunk 3]
```diff
@@ -85,9 +112,48 @@ declare namespace zipkin {
...
+      id:             string;
+      traceId:        option.IOption<string>;
+      spanId:         string;
+      name:           string;
+      kind:           string;
+      timestamp:      number;
+      duration:       number;
+      localEndpoint:  Endpoint;
+      remoteEndpoint: Endpoint;
+      annotations:    any[];
```
Comment (adriancole) [inline packages/zipkin/index.d.ts]: we could consider exporting model.Annotation I guess..

File: packages/zipkin/index.d.ts (hunk 2 above)
Comment (pbadenski) [reply packages/zipkin/index.d.ts]: yes: ```{ [key: string]: string }```. Will do that.

File: packages/zipkin/index.d.ts (hunk 3 above)
Comment (pbadenski) [reply packages/zipkin/index.d.ts]: Makes sense. Will do.

File: packages/zipkin/index.d.ts (hunk 1 above)
Comment (pbadenski) [reply packages/zipkin/index.d.ts]: oh, ipv4 is a string (of a "X.X.X.X" format)?

**Comment by adriancole** [comment]
Yep this model type matches exactly the openapi/swagger def which is ip
string literal. We resolve from InetAddress type prior.

File: packages/zipkin/index.d.ts [hunk 4]
```diff
@@ -117,12 +117,17 @@ declare namespace zipkin {
...
       setServiceName(serviceName: string): void;
-      setIpv4(ipv4: InetAddress): void;
+      setIpv4(ipv4: string): void;
       setPort(port: number): void;
 
       isEmpty(): void;
     }
 
+    interface Annotation {
+      timestamp: string;
```
Comment (adriancole) [inline packages/zipkin/index.d.ts]: number, right?

**Comment by adriancole** [comment]
rebased and force pushed.. PTAL

**Comment by pbadenski** [comment]
👍 Looks good to me. Thanks!

**Pull request merged**